src/commands/generate/__tests__/resources/readonly-test-cases/example-service/product-datasets/backingFileSystem.ts

```diff
@@ -0,0 +1,8 @@
+export interface IBackingFileSystem {
```

Comment:
I'm curious what's the reason for marking `configuration` as readonly, but not `fileSystemId` and `baseUri` (or even `IBackingFileSystem`)?

Reply:
Oh I see this goes together with `readonlyInterfaces`. I guess my question then becomes - should we have two separate flags for both or just one?

Reply:
I'm happy to collapse the two concepts into one flag, perhaps `readonlyInterfaces`. In which case I will merge #220 in to this and collapse the options

Reply:
Done, and added to the readme